### DISCUSSION TITLE
Why does DXVK not work for DirectX9 games?

### DISCUSSION THREAD

**Discussion by athei** [The-Wineskin-Project/WineskinServer#460] [Q&A] [answered]
DXVK does support DirectX9 according to their github README. Why does the WineskinServer README state that it is only used for DirectX9 + 10?

I am on an M1 Ultra macOS Sonomoa and trying to run the orignal WoW vanilla (32bit DX9 game). It has 9fps. So I assume it emulates DX9 via OpenGL?

**Comment by Gcenx** [comment, marked as answer]
Because MoltenVK doesn't provide some required extensions making it useless for us currently, this is also why CrossOver even the latest version doesn't make use of dx9.dll on macOS.

**Reply by athei** [reply]
Can you elaborate which extensions specifically are missing for DX9 support? Here it says that 1.10.3 only needs Vulkan 1.1: https://github.com/doitsujin/dxvk/wiki/Driver-support

It doesn't say anything about the difference requirements between DX9 and DX10/11. So I guess this information is hidden somewhere else.

**Reply by Gcenx** [reply]
DXVK really requires Vulkan 1.1 with many optimal extensions that we don’t have access to via MoltenVK currently.

Some of these can easily be ignored for dx10/dx11 but for dx9 most games I’d tried simply rendered as a black screen.

DXVK-macOS simply allows games to function without some of these additional extensions but that’s also why some have strange rendering issues. This is also why upstream DXVK never officially supported macOS.

**Reply by athei** [reply]
Thanks for clarifying.
